hips/hip-00NN.md

```diff
@@ -0,0 +1,101 @@
+
+## Specification
+
+The `Chart.yaml` should support the following format for `dependencies`:
+
+```
+dependencies:
+- name: "<dependency name>"
+  repository: "<protocol>://[<user>[:<password>]@]<hostname>[:<port>][:][/]<path>"
+  version: "<commit-ish>"
+```
+where:
+- `<protocol>` is one of `git`, `git+ssh`, `git+http`, `git+https`, or `git+file`.
+- `<commit-ish>` is an existing reference (SHA hash, tag or branch name) on the repo.
+
+For example:
+
+```
+dependencies:
+- name: "jenkins"
+  repository: "git+https://github.com/jenkinsci/helm-charts.git/charts/jenkins"
+  version: "main"
+```
+
+When Helm is installing a dependency from git, it should:
+
+- create a temporary directory
+- clone the repo at the specified branch/tag into the temp dir
+- treat the cloned git repo similar to a `file:///path/to/temp/dir` style requirement; use `chart.LoadDir` to load that directory (which in turn applied the logic for filtering the files through `.helmignore`) and archives it to `charts/`
+- delete the temp dir
+
+## Backwards compatibility  
+
+This is backwards compatible from Helm perspective - the existing formats for `dependencies` are still supported.
+
+Charts that start using the new format will effectively be changing their minimum required Helm version, i.e. they would be introducing breaking changes and should bump their major version.
+
+## Security implications
+
+Pulling the dependencies from git may introduce additional attack surfaces, as it would need to rely on an implementation of `git` (most likely the official `git` executable), and there have been recent vulnerabilities disclosed, including Remote Code Execution (RCE).
+
+This is something that needs to be taken into account in security conscious environments and might need to be documented for the end users. Users with high security requirements, should probably avoid using the feature and instead rely on a registry.
+
+## How to teach this
+
+- The documentation should note the security caveat listed above
+- The documentation should provide the recommendation to prefer registries to git, if possible
+- The documentation should note the implications of git being mutable with a recommendation of pinning to specific hashes
+
+## Reference implementation
+
+Multiple implementation attempts available for [a discarded earlier draft of a related HIP](https://github.com/helm/community/pull/214):
+
+- [helm/helm#11258](https://github.com/helm/helm/pull/11258)
+- [helm/helm#9482](https://github.com/helm/helm/pull/9482)
+- [helm/helm#6734](https://github.com/helm/helm/pull/6734)
+
+## Rejected ideas
+
+- An [earlier draft for solving the issue](https://github.com/helm/community/pull/214) suggested using URLs like `git://https://...`. There were comments about that approach in the reference implementations, with the suggestion to use the conventions which are already established in other ecosystems.
```

Comment:
agree, the established convention (as mentioned appears to be e.g `git+https://...`